fe/lib/components/JobCategorySuggest/JobCategorySuggest.tsx

```diff
@@ -0,0 +1,89 @@
+import { useLazyQuery } from '@apollo/react-hooks';
+import ApolloClient from 'apollo-client';
+import { FieldMessage, Loader, Stack, Text } from 'braid-design-system';
+import React, { forwardRef, useEffect } from 'react';
+import { useDebounce } from 'use-debounce';
+
+import {
+  JobCategorySuggestionChoice,
+  JobCategorySuggestionPositionProfileInput,
+} from 'lib/types/seek.graphql';
+
+import JobCategorySuggestChoices from './JobCategorySuggestChoices';
+import { JOB_CATEGORY_SUGGESTION } from './queries';
+
+interface Props {
+  client?: ApolloClient<unknown>;
+  schemeId: string;
+  debounceDelay?: number;
+  positionProfile: JobCategorySuggestionPositionProfileInput;
+  onSelect?: (jobCategorySuggestionChoice: JobCategorySuggestionChoice) => void;
+  showConfidence?: boolean;
+}
+
+export const JobCategorySuggest = forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      schemeId,
+      onSelect,
+      client,
+      positionProfile,
+      debounceDelay = 250,
+      showConfidence,
+    },
+    forwardedRef,
+  ) => {
+    const [
+      getCategorySuggestion,
+      { data: suggestData, error: suggestError, loading: suggestLoading },
+    ] = useLazyQuery(JOB_CATEGORY_SUGGESTION, {
+      client,
+      fetchPolicy: 'no-cache',
+    });
+
+    const [debounceJobCategorySuggestInput] = useDebounce(
+      positionProfile,
+      debounceDelay,
+    );
+
+    useEffect(() => {
+      if (debounceJobCategorySuggestInput) {
+        getCategorySuggestion({
+          variables: {
+            positionProfile: debounceJobCategorySuggestInput,
+            schemeId,
+            first: 5,
+          },
+        });
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [schemeId, debounceJobCategorySuggestInput]);
```

Comment:
Same thing as before, listening to "getCategorySuggestion" function for updates caused extraneous renders.

Reply:
I guess ESLint encourages us to memoise everything? Doesn't seem worth the effort.

Reply:
Yeah, I was going to look into eventually using `useMemo`, however at this stage, this was mostly a migration into wingman. I will eventually go back and update these setting a nicer pattern moving forward.